tests/test_auth/TestControllers.py

```diff
@@ -0,0 +1,109 @@
+import time
+
+from tests.base import BaseTestCase, SignUpInformation
+from flask import url_for
+from mod_auth.controllers import generate_hmac_hash
+
+
+class TestSignUp(BaseTestCase):
+
+    def test_if_email_signup_form_renders(self):
+        response = self.app.test_client().get(url_for('auth.signup'))
+        self.assertEqual(response.status_code, 200)
+        self.assert_template_used('auth/signup.html')
+
+    def test_blank_email(self):
+        response = self.signup(email='')
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(b'Email address is not filled in', response.data)
+
+    def test_invalid_email_address(self):
```

Comment:
This should preferably be done with an iterator feature of the test framework, so that each email is a separate test-case.

Reply:
Done.